teps/0150-displayname-in-matrix.md

```diff
@@ -0,0 +1,262 @@
+    "name": "matrixed-pr-vcx79-platforms-and-browsers-1",
+    "pipelineTaskName": "platforms-and-browsers"
+  }
+]
+```
+
+The `spec.tasks[].displayName` was introduced in [TEP-0047](0047-pipeline-task-display-name.md) which originally allowed
+any human-readable plain text and later [enhanced](https://github.com/tektoncd/pipeline/pull/7273) to support using
+`params` to allow runtime information in the `displayName`. This proposal is extending the support for the `displayName`
+such that the fully resolved `displayName` will be made part of the `pipelineRun.status.childReferences` whenever
+specified.
+
+### `tasks[].displayName` and `tasks[].matrix.include[].name`
+
+`matrix.include[]` section allows specifying a `name` along with a list of `params`. The `name` field has no clear
+purpose in terms of utilization. We propose to further enhance the `name` field and make it available as part of the
+`pipelineRun.status.childReferences[].displayName` if specified. `spec.tasks[].displayName` and
+`spec.tasks[].matrix.include[].name` can co-exist but `spec.tasks[].matrix.include[].name` takes higher precedence. We
+propose `spec.tasks[].matrix.include[].name` is included in the `pipelineRun.status.childReferences` and
+`pipelineRun.status.childReferences[].displayName` is set to `spec.tasks[].matrix.include[].name`. Along with this
+proposal, the pipeline author can specify `params` in `spec.tasks[].matrix.include[].name` which are resolved in the
+`childReferences`.
+
+```yaml
+- name: platforms-and-browsers-with-include
+  matrix:
+    include:
+      - name: "Platform: $(params.platform)"
+        params:
+          - name: platform
+            value: linux111
+  params:
+    - name: browser
+      value: chrome
+- name: platforms-and-browsers-with-include-but-not-resolved
+  matrix:
+    include:
+      - name: "Platform: $(params.browser)"
+        params:
+          - name: platform
+            value: linux111
+  params:
+    - name: browser
+      value: chrome
+```
+
+### Precedence Order
+
+| specification                                                                            | precedence                      | `childReferences[].displayName` |
+|------------------------------------------------------------------------------------------|---------------------------------|---------------------------------|
+| `tasks[].displayName`                                                                    | `tasks[].displayName`           |                                 |
+| `tasks[].matrix.include[].name`                                                          | `tasks[].matrix.include[].name` | `tasks[].matrix.include[].name` |
+| `tasks[].displayName` and `tasks[].matrix.include[].name`                                | `tasks[].matrix.include[].name` | `tasks[].matrix.include[].name` |
+
+### Notes and Caveats
+
+- This proposal is introducing an additional field in `childReferences` which can potentially increase the overall size
```

Comment:
Might be a NIT implementation detail but I am wondering if there is gonna be any validation of the length of display-name alike the 63 len limit for PipelineTaskName?

(doesn't need to block the merge of this)

Reply:
Yup, there are chances of having an extra large `displayName`. But since this is an opt-in feature, we can leave the validation part for now and revisit if needed in future.